js/coinsbit.js

```diff
@@ -0,0 +1,682 @@
+                    // '30020': BadSymbol, // Restricted symbol, API access is not allowed for the time being
+                    // '30021': BadSymbol, // Invalid symbol
+                    // '33333': BadSymbol, // {"code":33333,"msg":"currency can not be null"}
+                },
+                'broad': {
+                },
+            },
+        });
+    }
+
+    checkExchangeType (methodName, params) { // This method can be moved into base
+        // My proposal is to use separate exchange-type (spot|margin|futures) for each API call. So, we can assume a default to be always 'spot' but typically, in every function we should have created a separate parameter, named 'exchangeType', instead of generic 'type', because 'type' can be a very specific to the individual method, and it's better that there was a specific variable name dedicated for 'exchange-type' meaning. Imho, that shouldn't be 'type' but 'exchangeType'(or alike).
+        const defaultType = this.safeString2 (this.options, methodName, 'defaultType', 'spot');
+        const exchangeType = this.safeString (params, 'exchangeType', defaultType);
+        const supportedExchangeTypes = Object.keys (this.api);
+        if (!supportedExchangeTypes.includes (exchangeType)) {
+            throw new ExchangeError (this.id + " does not support '" + exchangeType + "' type, set exchange.options['defaultType'] to desired: " + supportedExchangeTypes.join (' | ')); // eslint-disable-line quotes
+        }
+        params = this.omit (params, 'exchangeType');
+        return [exchangeType, params];
+    }
+
+    safeMarketObject (element) { // This method can be moved into base
+        // I see there are many new exchange classes (mexc, gateio, ...) where inside 'fetchMarkets' these properties are being set for spot-market objects (on each object) while in original exchange API responses those properties doesn't exist at all. So, these can be filled with this helper function (like 'safeTicker' does for each ticker object)
+        return this.deepExtend (element, {
+            'spot': undefined,
+            'derivative': undefined,
+            'contract': undefined,
+            'linear': undefined,
+            'inverse': undefined,
+            'contractSize': undefined,
+            'expiry': undefined,
+            'expiryDatetime': undefined,
+            'strike': undefined,
+            'optionType': undefined,
+            'settle': undefined,
+            'settleId': undefined,
+        });
+    }
+
+    validateTimestamp (methodName, ts) { // This method can be moved into base
+        // Validate if provided timestamp is in MILLISECONDS format (13 digits), and not SECONDS (10 digits)
+        if (ts && ts.toString ().length !== 13) {
+            throw new ExchangeError ('Provided timestamp to method "' + methodName + '" should be in Milliseconds (13 digit).');
+        }
+    }
+
+    async fetchStatus (params = {}) {
```

Comment:
Let's not emulate it with a fetchMarkets-endpoint, if they don't have a specific endpoint for an exchange status. The less emulation – the better. We just implement what they truly have.

Reply:
Ah, i thought more emulation better :) [better-than-nothing]
however, sure, I'll remove that now.